### ISSUE TITLE
merge code of fork https://github.com/IngoZenz/ics-openvpn or provide similar capability

### ISSUE THREAD

**Issue by dsnz** [schwabe/ics-openvpn#1457]
Hi,
this excellent OpenVpn implementation is one of the most used OpenVpn Android clients 
it's also updated which is important for security reasons too

however Android allows only one VPN app active at anytime and certain networking related apps must work as a VPN implementation unless they have root access so it's impossible to have a normal VPN app active at the same time as another useful app app that must be implemented as a VPN app

one such app is personal DNS filter for which a fork of this app (OpenVPN for Android) has been created that makes available a DNS proxy to be used by personal DNS filter app
but this fork is not updated and it's an older version

I think this capability to make OpenVpn possible to interface with other networking apps is important and useful to users

I propose that the changes in  https://github.com/IngoZenz/ics-openvpn  are merged within this code as a selectable capability or otherwise provide at least the capability to log DNS requests (in a separate log)

thank you

**Comment by schwabe** [comment]
Pull requests should come as Pull request. For two reasons. One merging with that repo is really hard to review and second I so far have always required that I keep the copyright and right to relicense. Taking just GPL2 code from a fork would invalidate that for me.

The app already has a API for external adds and it is entirely possible to add functionality allow intercepting the tun fd with a pipe that goes to another app. For a DNS app it might be easier to just set the DNS server.

Basically I need a proper description what the other app needs and then I can look into it.

**Comment by dsnz** [comment]
Hi,
thanks for replying promptly

I have no relation with the other project and if they violate your license

I just made a proposal from the end-user perspective and I do not also understand if the api you mentioned should be fine for the other app

the other app is

https://github.com/IngoZenz/personaldnsfilter

and is doing two things, logging all dns requests and blocking (filtering) some of them
but just the logging is okay for most people
(it's an anti-malware capability as you understand)

I can't really enter in more discussion in this thread as I am posting here as an end-user without internal knowledge

I just hope that some solution allows the use of the other app while your openvpn is in use
(and if you believe the api you mentioned is enough then I will pass your comment to the other project)
or just dns logging be implemented within your app

thank you :)

**Comment by schwabe** [comment]
No they don't violate the license. I am just saying I want to keep full ownership of ics-openvpn and want to be able to license my own project under something else if I want to and requires the submitter to allow me to do that.

The current API is probably not enough. I am just saying that I am open to extending the API
